pure.zsh

```diff
@@ -57,17 +59,28 @@ prompt_pure_clear_screen() {
 }
 
 prompt_pure_check_git_arrows() {
+	# reset git arrows
+	prompt_pure_git_arrows=
+
 	# check if there is an upstream configured for this branch
 	command git rev-parse --abbrev-ref @'{u}' &>/dev/null || return
 
-	local right left arrows
-	right=$(command git rev-list --right-only --count HEAD...@'{u}' 2>/dev/null)
-	left=$(command git rev-list --left-only --count HEAD...@'{u}' 2>/dev/null)
+	# check arrow status inside single subshell for performance
+	local output
+	output="$(
+		command git rev-list --right-only --count HEAD...@'{u}' 2>/dev/null &&
+		command git rev-list --left-only --count HEAD...@'{u}' 2>/dev/null
+	)"
+	# exit if either command failed
+	(( !$? )) || return
+
+	# split output on newlines and read right / left
+	local arrows right=${${(@f)output}[1]} left=${${(@f)output}[2]}
```

Comment:
What's the benefit of doing it like this? Seems a bit convoluted to run the command, combine them, and then split.